src/components/DatatableV2/Table/Table.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+import styled from 'styled-components';
+import {
+  CellProps,
+  Column,
+  IdType,
+  SortingRule,
+  useFlexLayout,
+  usePagination,
+  useRowSelect,
+  useSortBy,
+  useTable,
+} from 'react-table';
+import { keys, prop, F as stubFalse } from 'ramda';
+import cls from 'classnames';
+import { isNonEmptyArray, isNotUndefined, isString } from 'ramda-adjunct';
+
+import { DatatableStore } from '../Datatable.store';
+import { tableActionsReducer } from './Table.reducer';
+import { actionsColumn } from './columns/actionsColumn';
+import { selectionColumn } from './columns/selectionColumn';
+import { Head } from './Head';
+import { Body } from './Body';
+import { Footer } from './Footer';
+import StyledTable from './Table.styles';
+import CellRenderer from './Body/renderers/CellRenderer';
+import { FlexContainer } from '../../FlexContainer';
+import { getColor, pxToRem } from '../../../utils/helpers';
+import { TableProps } from './Table.types';
+
+const NoDataContainer = styled(FlexContainer).attrs(() => ({
+  flexDirection: 'column',
+}))`
+  background: ${getColor('graphite5H')};
+  padding: ${pxToRem(64)};
+`;
+
+function collectFetchParams<D>(
+  pageIndex: number,
+  pageSize: number,
+  sortBy: SortingRule<D>[],
+): void {
+  DatatableStore.update((s) => {
+    s.pageIndex = pageIndex;
+    s.pageSize = pageSize;
+    s.sortBy = sortBy;
+  });
+}
+function collectSelectedIds<D>(selectedRows: Record<IdType<D>, boolean>): void {
+  DatatableStore.update((s) => {
+    s.selectedIds = keys(selectedRows);
+  });
+}
+
+const Table = <D extends Record<string, unknown>>({
+  columns,
+  data,
+  dataPrimaryKey,
+  dataSize,
+  isDataLoading,
+  rowActions,
+  NoMatchingDataComponent,
+  NoDataComponent,
```

Comment:
Both the default components here are taken from `defaultTableConfig` but they are optional props. Perhaps better to move the default values here?